components/Searchbar.js

```diff
@@ -0,0 +1,151 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useRef, useState, useEffect } from 'react';
+
+export default function Searchbar({ books }) {
+	// Hooks for search bar
+	const searchRef = useRef(null);
+
+	// router
+	const router = useRouter();
+		
+	/**
+	 * @param {HTMLElement} child 
+	 * @param {HTMLElement} parent 
+	 */
+	/*
+	const isDeepChildOf = (child, parent) => {
+		console.log(child.parentNode, parent);
+		if (child == parent) {
+			return true;
+		} else if (child === document.body || child === document.documentElement || document.head || window) {
+			return false;
+		} else {
+			return isDeepChildOf(child.parentNode, parent);
+		}
+	}
+	*/
+
+	useEffect(() => {
+		window.addEventListener('keydown', (e) => {
+			if (e.key === '/' || (e.key === 'k' && e.ctrlKey)) {
+				if (searchRef.current && e.target !== searchRef.current) {
+					e.preventDefault();
+					searchRef.current.focus();
+				}
+			} else if (e.key === 'Escape') {
+				searchRef.current.value = '';
+				searchRef.current.blur();
+				setSearchItems('hidden');
+				setSelected(null);
+				setInputValue('');
+			}
+		});
+	}, []);
+
+	// list of books to be shown under search
+	const [searchItems, setSearchItems] = useState('hidden');
+	// selected book in list
+	const [selected, setSelected] = useState(null);
+	// current value of input
+	const [inputValue, setInputValue] = useState('');
+
+	return (
```

Comment:
This file is using tabs. They should be spaces in increments of 2.

Reply:
can't get the spaces to work?

Reply:
You are using VS Code right? We can discuss this on Wednesday.